### PR TITLE
[SPARK-31355][SQL][DOCS] Document TABLESAMPLE in SQL Reference

### PR DESCRIPTION
**Pull request by huaxingao**
### What changes were proposed in this pull request?
Document TABLESAMPLE in SQL Reference

### Why are the changes needed?
To make SQL Reference complete

### Does this PR introduce any user-facing change?
Yes

Screen Shot 2020-04-06 at 10 23 52 PM

Screen Shot 2020-04-06 at 10 24 26 PM

### How was this patch tested?
Manually build and check.

### CODE REVIEW

**Comment by huaxingao** [comment]
@maropu A couple of more PRs to review. Thanks!

**Comment by maropu** [comment]
@huaxingao Yea, thanks for the great work!

File: docs/sql-ref-syntax-qry-sampling.md [hunk 1]
```diff
@@ -19,4 +19,72 @@ license: |
...
 
-**This page is under construction**
+### Description
+
+The `TABLESAMPLE` statement is used to sample the table. It supports the following sampling methods:
+- `TABLESAMPLE`(x `ROWS`): Sample the table down to the given number of rows.
+- `TABLESAMPLE`(x `PERCENT`): Sample the table down to the given percentage. Note that percentages are defined as a number between 0 and 100.
+- `TABLESAMPLE`(`BUCKET` x `OUT OF` y): Sample the table down to a `x` out of `y` fraction.
+
+Note: `TABLESAMPLE` returns the approximate number of rows or fraction requested.
```
Comment (maropu) [inline docs/sql-ref-syntax-qry-sampling.md:29]: We need a link to this section from the `SELECT` document?
https://github.com/apache/spark/blame/master/docs/sql-ref-syntax-qry-select.md#L78-L85

In the PostgreSQL doc, the `SELECT` document includes the `TABLESAMPLE` clause: https://www.postgresql.org/docs/current/sql-select.html

Comment (huaxingao) [reply docs/sql-ref-syntax-qry-sampling.md:29]: Thanks for your comments. 
In the related statement, I will cross link ```select``` and ```tablesample```. I am a little hesitate to add tablesample to the ```from item```.  Seems to me, Tablesample is not parallel to Table relation or InlineTable, the structure is more like this:
```
SELECT ... FROM 
    Table relation
       Tablesample
    ......
    Inlined table
       Tablesample
```
Here is how snowflake document this https://docs.snowflake.com/en/sql-reference/constructs/from.html

Comment (maropu) [reply docs/sql-ref-syntax-qry-sampling.md:29]: ok, thanks for the check. Looks ok as it is.

**Review by maropu — APPROVED** [review]
cc: @srowen

File: docs/sql-ref-syntax-qry-sampling.md [hunk 2]
```diff
@@ -19,4 +19,74 @@ license: |
...
+    | TABLESAMPLE (BUCKET integer_expression OUT OF integer_expression)
+{% endhighlight %}
+
+### Examples
+{% highlight sql %}
+SELECT * FROM test;
++--+----+
+|id|name|
++--+----+
+|5 |Alex|
```
Comment (maropu) [inline docs/sql-ref-syntax-qry-sampling.md:45]: btw, why does the result is left-aligned? The output of `showString` seems to be right-aligned?
```
scala> sql("SELECT * FROM test;").show()
+---+----+
| id|name|
+---+----+
|  1| aaa|
+---+----+
```

**Comment by srowen** [comment]
Jenkins retest this please

**Comment by srowen** [comment]
Looks OK just may need a rebase @huaxingao

**Comment by srowen** [comment]
Merged to master/3.0

**Comment by huaxingao** [comment]
Thanks! @maropu @srowen

**Pull request closed without merging**